guarddog/scanners/scanner.py

```diff
@@ -233,35 +233,31 @@ def scan_local(
         Args:
             path (str): path to package
             rules (set, optional): Set of rule names to use. Defaults to all rules.
+            callback (typing.Callable[[dict], None], optional): Callback to apply to Analyzer output
 
         Raises:
             Exception: Analyzer exception
 
         Returns:
             dict: Analyzer output with rules to results mapping
-            rules: rules to apply
-            callback: callback to call for each result
         """
 
         if rules is not None:
             rules = set(rules)
 
-        if not os.path.exists(path):
-            raise Exception(f"Path {path} does not exist.")
-
-        if any(path.endswith(ext) for ext in (".tar.gz", ".tgz", ".zip", ".whl")):
-            with tempfile.TemporaryDirectory() as tmpdirname:
-                safe_extract(path, tmpdirname)
-                return self.analyzer.analyze_sourcecode(
-                    tmpdirname, rules=rules
-                )
-
+        results = None
         if os.path.isdir(path):
-            return self.analyzer.analyze_sourcecode(path, rules=rules)
+            results = self.analyzer.analyze_sourcecode(path, rules=rules)
+        elif os.path.isfile(path):
+            with tempfile.TemporaryDirectory() as tempdir:
+                safe_extract(path, tempdir)
+                results = self.analyzer.analyze_sourcecode(tempdir, rules=rules)
+        else:
+            raise Exception(f"Local scan target {path} is neither a directory nor a file.")
```

Comment:
I think it's slightly more correct without the reference to "supported."

`PackageScanner.scan_local()` now defers the check for supported archive to `safe_extract()` (line 253), so if there's an unsupported archive error, it will come up there.  At this point in `scan_local()`, it's just checking for directory or file.

This case is kind of strange besides because, as long as you only call `PackageScanner.scan_local()` via `cli._scan()`, this error condition is never reached.  I left it in just for the sake of completeness.

Reply:
I just realized the error message from `safe_extract()` isn't correctly printing the name of the incompatible archive file.  I'll add this to my commit with the improvements in `tests/`

Reply:
Continuing where my first comment left off, part of the appeal of changing the behavior of the local directory scan would mean that we can make it so `PackageScanner.scan_local()` always expects a path to a directory, so no more having to check for directory, file or error